xdsl/ir/affine/affine_expr.py

```diff
@@ -37,6 +39,32 @@ def dimension(position: int) -> AffineExpr:
     def symbol(position: int) -> AffineExpr:
         return AffineSymExpr(position)
 
+    @staticmethod
+    def binary(
+        kind: AffineBinaryOpKind,
+        lhs: AffineExpr,
+        rhs: AffineExpr,
+    ) -> AffineExpr:
+        """
+        The difference between creating an `AffineBinaryOpExpr` with this function
+        and creating an `AffineBinaryOpExpr` using its constructor
+        is that simplifications are automatically applied in this function.
+        """
```

Comment:
Nit: can you please first state what this method does, and then elaborate on the differences with the aforementioned constructor?

The text you have already there for the latter is fine, but for the uninitiated it might be good mentioning when does this simplification take place (e.g., is it when calling `eval` for example?).